src/components/chart/brush/index.js

```diff
@@ -0,0 +1,204 @@
+import React, { PureComponent } from 'react';
+
+import { scaleLinear } from 'd3-scale';
+import PropTypes from 'prop-types';
+
+import SVGBrush from './component';
+
+export default class Brush extends PureComponent {
+  static propTypes = {
+    width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    startIndex: PropTypes.number,
+    endIndex: PropTypes.number,
+    margin: PropTypes.object,
+    data: PropTypes.array,
+    minimumGap: PropTypes.number,
+    maximumGap: PropTypes.number,
+    onBrushEnd: PropTypes.func,
+  };
+
+  static defaultProps = {
+    margin: {
+      top: 0,
+      left: 0,
+      bottom: 0,
+      right: 0,
+    },
+    minimumGap: 0,
+    maximumGap: 0,
+  };
+
+  state = {
+    ready: false,
+    animate: false,
+    brushSelection: null,
+  };
+
+  componentDidMount() {
```

Comment:
I get you saw this code somewhere and it will just work, but it's using React's old lifecycle. Bringing this to a refactor is like shooting in your own foot.